backend/pkg/kubeconfig/kubeconfig.go

```diff
@@ -316,299 +335,362 @@ func loadContextsFromData(data []byte, source int) ([]Context, []error) {
-					if err != nil {
-						return nil, DataError{Field: "certificate-authority-data", Reason: "invalid base64 encoding"}
-					}
-
-					cluster.CertificateAuthorityData = decoded
-				} else {
-					return nil, DataError{Field: "certificate-authority-data", Reason: "expected string, got different type"}
-				}
-			case "extensions":
-				cluster.Extensions = createExtensions(value)
-			default:
-				err := SetClusterField(cluster, strKey, value)
-				if err != nil {
-					return nil, ClusterError{ClusterName: clusterName, Reason: err.Error()}
-				}
-			}
+	clusterName, ok := contextData["cluster"].(string)
+	if !ok {
+		return "", "", ContextError{
+			ContextName: contextName,
+			Reason:      fmt.Sprintf("missing or invalid cluster name: %v", contextData["cluster"]),
 		}
+	}
 
-		return cluster, nil
+	userName, ok := contextData["user"].(string)
+	if !ok {
+		return "", "", ContextError{
+			ContextName: contextName,
+			Reason:      fmt.Sprintf("missing or invalid user name: %v", contextData["user"]),
+		}
 	}
 
-	return nil, ClusterError{ClusterName: clusterName, Reason: "cluster not found"}
+	return clusterName, userName, nil
 }
 
-// findAuthInfo finds an auth info in the given raw users.
-// It returns the found auth info and any errors that occurred.
-// It returns an error if the auth info cannot be found.
-// It returns a valid auth info if the auth info is found.
-// It returns any errors that occurred during auth info search.
-func findAuthInfo(rawUsers []interface{}, userName string) (*api.AuthInfo, error) {
-	for _, rawUser := range rawUsers {
-		userMap, ok := rawUser.(map[interface{}]interface{})
-		if !ok || userMap["name"] != userName {
-			continue
+// getClusterAndUser gets the cluster and user details from the kubeconfig.
+func getClusterAndUser(
+	kubeconfig map[string]interface{},
+	clusterName,
+	userName string,
+) (map[interface{}]interface{}, map[interface{}]interface{}, error) {
+	cluster, err := getCluster(kubeconfig, clusterName)
+	if err != nil {
+		return nil, nil, ContextError{
+			ContextName: clusterName,
+			Reason:      fmt.Sprintf("error getting cluster: %v", err),
+		}
+	}
+
+	user, err := getUser(kubeconfig, userName)
+	if err != nil {
+		return nil, nil, ContextError{
+			ContextName: userName,
+			Reason:      fmt.Sprintf("error getting user: %v", err),
 		}
+	}
 
-		authInfo := &api.AuthInfo{}
-		userData, _ := userMap["user"].(map[interface{}]interface{})
+	return cluster, user, nil
+}
 
-		for key, value := range userData {
-			strKey, ok := key.(string)
-			if !ok {
-				return nil, UserError{UserName: userName, Reason: fmt.Sprintf("invalid key type for field '%v'", key)}
-			}
+// createAndValidateConfig creates and validates the config.
+func createAndValidateConfig(
+	contextName string,
+	contextMap,
+	cluster,
+	user map[interface{}]interface{},
+	kubeconfig map[string]interface{},
+) (*api.Config, error) {
+	singleConfig := createKubeConfig(contextName,
+		toStringKeyMap(contextMap),
```

Comment:
why is the `toStringKeyMap` conversion used here?

Reply:
`toStringKeyMap` is used to ensure that all map keys are strings, which is a requirement for proper YAML marshaling and compatibility with the client-go library.

It ensures that all the key-value pairs in `contextMap, cluster, and user` have string keys before they're used to create the new config.

Reply:
@yolossn does this make sense for you?

Reply:
Yes it does